### ISSUE TITLE
Saving Best Model

### ISSUE THREAD

**Issue by samuelnaol72** [michaelneri/unsupervised-audio-anomaly-detection#2]
Hi Michael!!

Shouldn't we track the best model based on the minimum validation loss? It seems best model is being tracked at interval of 20 epochs?

![Image](https://github.com/user-attachments/assets/9711c475-4e37-4753-b59e-35a1c8c83111)

Thank you indvance!

**Comment by michaelneri** [comment]
Hi @samuelnaol72, 

I confirm that the model is tracked every 20 epochs based on the validation loss, but it is trained for 300-ish epochs. You can modify that 20 epochs and log it every epoch, I did not notice any differences when I was carrying out the experiments.

Michael
